tests/PHPStan/Rules/Arrays/OffsetAccessAssignmentRuleTest.php

```diff
@@ -127,4 +127,21 @@ public function testAssignNewOffsetToStubbedClass(): void
 		$this->analyse([__DIR__ . '/data/new-offset-stub.php'], []);
 	}
 
+
+	public function testRuleWithNullsafeVariant(): void
+	{
+		if (PHP_VERSION_ID < 80000) {
+			$this->markTestSkipped('Test requires PHP 8.0.');
+		}
+
+		$this->checkUnionTypes = true;
+		// TODO: Does not work as expected
```

Comment:
TODO

Reply:
Thanks, I just though the rule is supposed to work differently at first but then I found the AppendedArrayKeyTypeRule and understood the difference :)

Both TODOs removed as the test works correctly.